File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>>result;
        if(root==NULL) return result;
        queue<TreeNode*>q;
        q.push(root);
        while(!q.empty()){
            vector<int>ans;
            int n=q.size();
            for(int i=0;i<n;i++){
                TreeNode *node=q.front(); // her ek element ko node bana kr check kar rahe hai
                q.pop();
                if(node->left != NULL) q.push(node->left);
                if(node->right != NULL) q.push(node->right);
                ans.push_back(node->val); // isko vector<vecor<int>> me daal do ...

            }
            result.push_back(ans);
        }
        return result;

        
    }
};

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        result = []
        if(root==None):
            return result
        q = deque()
        q.append(root)
        while (q):
            ans = []
            n = len(q)
            while(n):
                node = q[0]
                q.popleft()
                ans.append(node.val)
                if(node.left != None ):
                    q.append(node.left)
                if(node.right != None):
                    q.append(node.right)
                n-=1
            result.append(ans)
                
        return result 
                    





        
         
        